plugin/test/unit/templates/utils.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { tmpdir } from 'os'
+import { resolve, join, dirname } from 'path'
+
+import { copy, ensureDir, existsSync, readFileSync, unlink } from 'fs-extra'
+import { dir as getTmpDir } from 'tmp-promise'
+
+// eslint-disable-next-line import/no-namespace
+import * as templateUtils from '../../../src/templates/utils'
```

Comment:
This needs to be imported this way because of the way `downloadFile` needs to be stubbed. If we were to follow the same approach as seen in other test files, the `prepareFilesystem` method will retain the reference to the original, unstubbed version of `downloadFile` in the `prepareFilesystem` tests (which we don't want)

Reply:
Clever! I dread to think how long it took to track that one down